### PR TITLE
Chatbot Avatar Images

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
Ability to add avatar images for the "user" and "bot" in `gr.Chatbot()`. Simply provide image(s) `gr.Chatbot(avatar_images=["user_image", "bot_image"])`.

Screenshot 2023-08-18 at 12 03 33 PM

Please include a concise summary, in clear English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #3769

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by abidlabs** [comment]
Sorry @dawoodkhan82 pushed to this branch by accident -- will revert

**Comment by abidlabs** [comment]
Tested, changes look good! My only suggestion would be to use `self.make_temp_copy_if_needed()` and pass the file path to the frontend instead of passing in the base64. For large files, serialization can be slow, and we don't take advantage of the browser caching the file. 

Ref: https://stackoverflow.com/questions/1574961/how-much-faster-is-it-to-use-inline-base64-images-for-a-web-site-than-just-linki

**Comment by yvrjsharma** [comment]
Thanks, @dawoodkhan82 results look amazing. I passed none for user_image and idefics logo for the bot in this attached example. 
image

I was wondering if you plan to add avatar images to the _shared discussions_ as well. I can see that right now, it is not supported.
image

Sample Demo here - https://huggingface.co/spaces/ysharma/dummy_avatar

**Comment by dawoodkhan82** [comment]
@abidlabs Do you know how the shared discussions are generated?

**Comment by dawoodkhan82** [comment]
@abidlabs ready to merge after these tests pass.

**Comment by dawoodkhan82** [comment]
Want to confirm that `avatar_images` being an array and having an option to pass `None` is the best implementation right?

**Comment by abidlabs** [comment]
I would have probably picked tuple of 2 strings (tuples since its a fixed length). Allowing `None` is great!

**Comment by yvrjsharma** [comment]
Allowing None was genius tbh!

File: js/chatbot/static/ChatBot.svelte [hunk 1]
```diff
@@ -42,6 +44,9 @@
 
 	let div: HTMLDivElement;
 	let autoscroll: boolean;
+	let images_dir: string = root_url
```
Comment (dawoodkhan82) [inline js/chatbot/static/ChatBot.svelte:47]: @abidlabs Isn't this what `normalise_files` is meant to do?

**Comment by abidlabs** [comment]
LGTM @dawoodkhan82! Among other things, I confirmed:

(1) That this works with `.update()`

```py
import gradio as gr

with gr.Blocks() as demo:
    c = gr.Chatbot([("abc", "def")], avatar_images=["cat.jpg", None])
    btn = gr.Button()
    btn.click(lambda : gr.update(avatar_images=(None, "cheetah.jpg")), None, c)
    
demo.launch()
```

2. as well as with gr.ChatInterface:

```py
import gradio as gr

ch = gr.Chatbot(avatar_images=["cat.jpg", None])

gr.ChatInterface(lambda x,y:x, chatbot=ch).launch()
```

**Review by abidlabs — APPROVED** [review]

File: js/chatbot/static/ChatBot.svelte (hunk 1 above)
Comment (abidlabs) [reply js/chatbot/static/ChatBot.svelte:47]: Good point. I just took a look though and it looks like `normalise_files` doesn't handle the string case fully. In particular, it doesn't attach the `/file=` or `/proxy` prefixes if the input is a string.

I'm not sure about the full implications of patching `normalise_files` so I just added the logic there myself.

**Pull request merged**